src/System.Xml.XmlSerializer/tests/XmlSerializerTests.cs

```diff
@@ -1681,6 +1681,102 @@ public static void Xml_DeserializeOutOfRangeByteProperty()
         }
     }
 
+    [Fact]
+    public static void Xml_XmlAttributes_RemoveXmlElementAttribute()
+    {
+        XmlAttributes attrs = new XmlAttributes();
+
+        XmlElementAttribute item = new XmlElementAttribute("elem1");
+        attrs.XmlElements.Add(item);
+        Assert.True(attrs.XmlElements.Contains(item));
+
+        attrs.XmlElements.Remove(item);
+        Assert.False(attrs.XmlElements.Contains(item));
+    }
+
+    [Fact]
+    public static void Xml_XmlAttributes_RemoveXmlElementAttribute_ThrowsOnMissingItem()
+    {
+        XmlAttributes attrs = new XmlAttributes();
+
+        XmlElementAttribute item1 = new XmlElementAttribute("elem1");
+        attrs.XmlElements.Add(item1);
+
+        XmlElementAttribute item2 = new XmlElementAttribute("elem2");
+        attrs.XmlElements.Add(item2);
+        Assert.True(attrs.XmlElements.Contains(item1));
+        Assert.True(attrs.XmlElements.Contains(item2));
+
+        attrs.XmlElements.Remove(item2);
+        Assert.False(attrs.XmlElements.Contains(item2));
+
+        Assert.Throws<ArgumentException>(() => { attrs.XmlElements.Remove(item2); });
+    }
+
+    [Fact]
+    public static void Xml_XmlAttributes_RemoveXmlArrayItemAttribute()
+    {
+        XmlAttributes attrs = new XmlAttributes();
+
+        XmlArrayItemAttribute item = new XmlArrayItemAttribute("item1");
+        attrs.XmlArrayItems.Add(item);
+        Assert.True(attrs.XmlArrayItems.Contains(item));
+
+        attrs.XmlArrayItems.Remove(item);
+        Assert.False(attrs.XmlArrayItems.Contains(item));
+    }
+
+    [Fact]
+    public static void Xml_XmlAttributes_RemoveXmlArrayItemAttribute_ThrowsOnMissingItem()
+    {
+        XmlAttributes attrs = new XmlAttributes();
+
+        XmlArrayItemAttribute item1 = new XmlArrayItemAttribute("item1");
+        attrs.XmlArrayItems.Add(item1);
+
+        XmlArrayItemAttribute item2 = new XmlArrayItemAttribute("item2");
+        attrs.XmlArrayItems.Add(item2);
+        Assert.True(attrs.XmlArrayItems.Contains(item1));
+        Assert.True(attrs.XmlArrayItems.Contains(item2));
+
+        attrs.XmlArrayItems.Remove(item2);
+        Assert.False(attrs.XmlArrayItems.Contains(item2));
+
+        Assert.Throws<ArgumentException>(() => { attrs.XmlArrayItems.Remove(item2); });
+    }
+
+    [Fact]
+    public static void Xml_XmlAttributes_RemoveXmlAnyElementAttribute()
+    {
+        XmlAttributes attrs = new XmlAttributes();
+
+        XmlAnyElementAttribute item = new XmlAnyElementAttribute("elem1");
+        attrs.XmlAnyElements.Add(item);
+        Assert.True(attrs.XmlAnyElements.Contains(item));
+
+        attrs.XmlAnyElements.Remove(item);
+        Assert.False(attrs.XmlAnyElements.Contains(item));
+    }
+
+    [Fact]
+    public static void Xml_XmlAttributes_RemoveXmlAnyElementAttributeThrowsOnMissingItem()
```

Comment:
Yes they may overlap but to some extent that is not necessarily redundant and I don't see much harm with that. The current change will test the scenario where an item that is used to be in the collection being removed twice while there are other items in the collection as well whereas if we only have line 1777 then it would only test removing item that has never been added to the collection and the collection is empty at the time we remove the item. Given the current implementation there won't be any differences but these scenarios differ in some aspects. I think that these change will provide sufficient coverage for those scenarios.